app/scripts/views/mixins/settings-mixin.js

```diff
@@ -33,6 +33,17 @@ define([
         Session.clear();
         self.user.clearSignedInAccount();
       }
+    },
+
+    // Cache the profile image URL whenever it changes
+    // TODO: issue #2095 send broadcast
+    onAvatarChange: function (avatar) {
```

Comment:
Two comments:
* It seems like this functionality belongs in the avatar-mixin. 
* `onAvatarChange` hints that this function is called in response to an avatar change, on a model somewhere. The functionality is only kinda-sorta that, and also sets an avatar on the account and user. Maybe rename the function `updateAvatarUrl` and have it take just a URL?